pull request mongo-functions to master

gabr1gus opened the pull request:
--as funções solicitadas foram adicionadas e testadas, elas parecem estar funcionando corretamente

gabr1gus committed:
added mongoDB

--adicionado uma classe MongoConnection que ira lidar com as requisiçoes mongo
--agora a funçao send que siga o padrao adiciona uma instancia no mongo DB
--localhost:8000/app?command=send&sender=sender&receiver=receiver&message=corpo da mensagem&counter=contador
--aparentemente nao precisa criar um db ou uma tabela no banco de dados, mongo cuida disso para vc
--as bibliotecas foram adicionadas nas configuraçoes da intelij
--TODO: remover o log que aparece quando se seleciona um DB

gabr1gus committed:
created new funcions on mongoConnection

--add function that add message on the right collection
--add function that adds a new chat, this function MUST be called when a new chat is created
--add function that gets ALL the messages of the chat in the database as Array list of Document(basically Map)
--added test db files for mongo, this files are outdated and will be removed/altered later
--TODO: make a function that only gets the x to y messages from the mongoDb

gabr1gus committed:
new mongo functions

--variavel que determina o tamanho do lote de mensagens a serem puxadas do servidor por vez
--funcao que retorna a lista de usuarios
--NEED TEST:
--funcao que retorna o primeiro lote de mensagens
--funçao que retorna as proximas mensagens ao receber o document da utima mensagem na lista

GCastilho commented on src/database/data1.json:
```diff
@@ -0,0 +1,6 @@
+{"_id":{"$oid":"5ce9e5b2d96a795000229aa2"},"sender":"gabriel","receiver":"GCastilho","message":"mensagem 1"}
```
E o que seria esses arquivos data1.json e data 2.json e index.json? Arquivos para teste? Se sim, ele devem ser deletados e suas chamadas no código retiradas

GCastilho committed:
Minor fixes

--the filter in one of the mongoDb querry was "Greater than" instead of "lesser than"
--re implemented the function in the send command  that adds the message to the mongoDB

GCastilho commented on src/api/Api.java:
```diff
@@ -19,14 +23,26 @@ public static JSONObject process(JSONObject data) {
+				messageDoc.put("sender", sender);
+				messageDoc.put("receiver", receiver);
+				messageDoc.put("message", message);
+				messageDoc.put("time", data.getString("time"));
```
```suggestion
				messageDoc.put("timestamp", data.getString("timestamp"));
```

GCastilho commented on src/api/Api.java:
```diff
@@ -19,14 +23,26 @@ public static JSONObject process(JSONObject data) {
 				e.printStackTrace();
 				response.put("status", "error");
-				response.put("info", "Internal server error");
+				response.put("errorMessage", "Internal server error");
```
```suggestion
				response.put("info", "Internal server error");
```

GCastilho commented on src/api/Api.java:
```diff
@@ -1,10 +1,14 @@
 package api;
 
+import com.mongodb.*;
```
```suggestion

```

GCastilho commented on src/api/Api.java:
```diff
@@ -1,10 +1,14 @@
 import org.json.JSONObject;
 
 import java.io.IOException;
 import java.util.Arrays;
```
```suggestion

```

GCastilho reviewed (changes requested):
Vc tem q parar de ficar alterando código que não é pra vc alterar, cara
Além disso, tem q fazer rebase direto, fica uma zona qdo um de vcs fazem

GCastilho committed:
Implementado 'getUserList' na API

-- Implementado API call 'getUserList', que retorna a lista de usuários cadastrados no sistema
-- todos as mensagens 'ok' retornam um 'command' para o cliente saber que tipo de mensagem está sendo enviado
-- Minor fixes

gabr1gus committed:
Apply suggestions from code review

--minor change

Co-Authored-By: GCastilho <34466758+GCastilho@users.noreply.github.com>

gabr1gus committed:
added comments and minor chages

--removed the mongoDb creation files
--removed function that used to get all messages
--renamed some functions for a more fitting name

GCastilho reviewed (dismissed):
Coloca uma mensagem descritiva quando for dar merge, hein

gabr1gus committed:
modified getNextMessageBatch to receive 3 Strings

--removed the code that added a sender to the message Document on the mongoDatabase

gabr1gus committed:
modified updateCookie to just update the cookie timestamp

--acredito que essa função deveria ter um retorno porem não consigo lembrar qual o retorno que deveria ter, então deixei como void, 
--se for para retornar o user da para chamar getUser() apos essa função com o mesmo session id que terá o mesmo resultado,

GCastilho committed:
Adicionada 'getMessages' na API

-- Adicionada função 'getMessages' na API e implementada-a completamente, ela agora retorna a lista de mensagens para o cliente em um json
-- Minor fixes

GCastilho commented on src/api/Api.java:
```diff
@@ -35,7 +36,7 @@ static JSONObject process(JSONObject data) {
 				messageDoc.put("sender", sender);
 				messageDoc.put("message", message);
-				messageDoc.put("timestamp", data.getString("timestamp"));
+				messageDoc.put("timestamp", data.get("timestamp").toString());
```
O timestamp, no JSON **não é um string** (por isso q foi modificado de getString pra get->toString), então o melhor seria se o timestamp no mongo tbm não fosse uma string, mas um número, evitaria essa quase gambiarra que está sendo feita

GCastilho commented on src/api/Api.java:
```diff
@@ -19,17 +23,61 @@ public static JSONObject process(JSONObject data) {
+				Document messageDoc = new Document();
+				messageDoc.put("sender", sender);
+				messageDoc.put("message", message);
+				messageDoc.put("timestamp", data.get("timestamp").toString());
```
O timestamp, no JSON não é um string (por isso q foi modificado de getString pra get->toString), então o melhor seria se o timestamp no mongo tbm não fosse uma string, mas um número, evitaria essa quase gambiarra que está sendo feita

GCastilho reviewed (changes requested):
O timestamp no mongo deve ser mudado de string para (algum tipo de) número

gabr1gus committed:
Bug fix

--corrigido um bug que impedia que a api acessa-se as funcoes de send pois o session id era armazenado incorretamente no header e depois quando solicitado o mesmo estava errado em algum navegador

GCastilho committed:
API bugfix

-- Corrigido um bug na API que estava enviando 'responses' incorretamente

gabr1gus committed:
modified the tipe of data stored in the timeStamp field

--agora armazena como um in64
--embora exista um tipo de dado "timestamp" no db eu nao quero imaginar como diabos eu adiciono algo de tipo "timestamp"
--um comentario, timestamp é em teoria um campo desnecessario visto que algo como teoricamente document.get("_id").getTimestamp() deveria retornar o timestamp em que o document foi adicionado no mongoDatabase

GCastilho committed:
Minor API fix

-- Corrigido sender não estar sendo enviado quando a mensagem era enviada ao receptor
-- Corrigido response do 'send' não estar sendo enviado
